File: api/src/lib/types/domain/prayer-log.entity.ts
import type {
  PrayerCategory,
  PrayerMethod,
  PrayerPlace,
  PrayerType,
} from "../../generated/prisma/enums.js";

/**
 * Domain Entities
 * Representasi data dari database
 */
export interface IPrayerLog {
  id: string;
  user_id: string;
  date: Date;
  prayer: PrayerType;
  category: PrayerCategory;
  performed: boolean;
  performed_at: Date | null;
  method: PrayerMethod | null;
  place: PrayerPlace | null;
  is_qadha: boolean;
  notes: string | null;
  created_at: Date;
  updated_at: Date;
}

File: handlerclaw/lib/features/prayer-logs/domain/entities/prayer_log_entity.dart
class PrayerLogEntity {
  final String id;
  final String userId;
  final DateTime date;
  final String prayer;
  final String category;
  final bool performed;
  final DateTime performedAt;
  final bool isQadha;
  final String method;
  final String place;
  final String? notes;

  PrayerLogEntity({
    required this.id,
    required this.userId,
    required this.date,
    required this.prayer,
    required this.category,
    required this.performed,
    required this.performedAt,
    required this.isQadha,
    required this.method,
    required this.place,
    this.notes,
  });
}
